app/models/version.rb

```diff
@@ -28,9 +28,10 @@ class Version < ApplicationRecord # rubocop:disable Metrics/ClassLength
 
   validates :number, length: { maximum: Gemcutter::MAX_FIELD_LENGTH }, format: { with: Patterns::VERSION_PATTERN }
   validates :platform, length: { maximum: Gemcutter::MAX_FIELD_LENGTH }, format: { with: Rubygem::NAME_PATTERN }
-  validates :gem_platform, length: { maximum: Gemcutter::MAX_FIELD_LENGTH }, format: { with: Rubygem::NAME_PATTERN }, on: :create
-  validates :full_name, presence: true, uniqueness: { case_sensitive: false }
-  validates :gem_full_name, presence: true, uniqueness: { case_sensitive: false, allow_nil: true }, on: :create
+  validates :gem_platform, length: { maximum: Gemcutter::MAX_FIELD_LENGTH }, format: { with: Rubygem::NAME_PATTERN },
```

Comment:
this doesn't work. I think we might need to split this into two lines:

```ruby
validates :gem_platform, length: { maximum: Gemcutter::MAX_FIELD_LENGTH }, format: { with: Rubygem::NAME_PATTERN }, on: :create
validates :gem_platform, length: { maximum: Gemcutter::MAX_FIELD_LENGTH }, format: { with: Rubygem::NAME_PATTERN }, if: :gem_platform_changed?
```

since `on` doesn't work for method conditionals, that's only `if`, and we need the `on: :create` to catch the presence validation...

Reply:
or i guess

```ruby
validates :gem_platform, length: { maximum: Gemcutter::MAX_FIELD_LENGTH }, format: { with: Rubygem::NAME_PATTERN }, if: -> { validation_context == :create || gem_platform_changed? }
```